app/actions/websocket/channel.ts

```diff
@@ -1,21 +1,68 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {Model} from '@nozbe/watermelondb';
 import {DeviceEventEmitter} from 'react-native';
 
-import {localRemoveUserFromChannel, localSetChannelDeleteAt, switchToChannel} from '@actions/local/channel';
-import {updateUsersNoLongerVisible} from '@actions/remote/user';
+import {localRemoveCurrentUserFromChannel, localSetChannelDeleteAt, switchToChannel} from '@actions/local/channel';
+import {fetchChannelInfo, fetchMyChannel} from '@actions/remote/channel';
+import {fetchUsersByIds, updateNewUsersVisible, updateUsersNoLongerVisible} from '@actions/remote/user';
 import Events from '@constants/events';
 import DatabaseManager from '@database/manager';
 import {queryActiveServer} from '@queries/app/servers';
 import {deleteChannelMembership, queryCurrentChannel} from '@queries/servers/channel';
-import {queryConfig, setCurrentChannelId} from '@queries/servers/system';
+import {queryCommonSystemValues, queryConfig, setCurrentChannelId} from '@queries/servers/system';
 import {queryLastChannelFromTeam} from '@queries/servers/team';
-import {queryCurrentUser} from '@queries/servers/user';
+import {queryCurrentUser, queryUserById} from '@queries/servers/user';
 import {dismissAllModals, popToRoot} from '@screens/navigation';
 import {isTablet} from '@utils/helpers';
 import {isGuest} from '@utils/user';
 
+export async function handleUserAddedEvent(serverUrl: string, msg: any) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+    const {currentChannelId, currentUserId} = await queryCommonSystemValues(database.database);
+    const teamId = msg.data.team_id;
+    const channelId = msg.data.channel_id;
+    const userId = msg.data.user_id;
+
+    const models: Model[] = [];
+
+    try {
```

Comment:
I think the try/catch can be removed, but I will wait for feedback.

Reply:
if something in here throws we should keep it

Reply:
Any query to the database may throw an exception, so I will keep it.